lychee-lib/src/extract/markdown.rs

```diff
@@ -171,8 +178,22 @@ https://bar.com/123
 or inline like `https://bar.org` for instance.
 
 [example](http://example.com)
+
+<span id="the-end">The End</span>
         "#;
 
+    #[test]
+    fn test_extract_fragments() {
+        let expected = HashSet::from([
+            "a-test".to_string(),
+            "a-test-1".to_string(),
```

Comment:
This comes from that line, right?
```md
## A test {#well-still-the-same-test}
```

I was of the impression that the custom fragment name would overwrite the default fragment name? Does a host like GitHub generate both, `a-test-1` and `well-still-the-same-test`?

Reply:
Sorry, I didn't draw enough attention to this, briefly mentioned this in https://github.com/lycheeverse/lychee/pull/1126#issuecomment-1632970923.

GitHub ignores heading attributes.

If don't generate a kebab case heading when a heading attribute is present, then later kebab case headings may have incorrect numbers. So to minimize false negatives, we generate both if heading attributes are present.

Reply:
Maybe we can add a comment here to document the behavior.

Reply:
That's a good idea. Added a comment to the `extract_markdown_fragments` documenting this behaviour.